database/migrations/2017_05_20_130000_create_pages_table.php

```diff
@@ -32,6 +32,8 @@ public function up()
             $table->dateTime('published_at'); // Allows pages to be published at a later date.
             $table->softDeletes();
             $table->timestamps();
+
+            $table->unique(['slug', 'parent_id']);
```

Comment:
Duplicate page slugs are OK, provided they have different parents.  Real example - its certainly possible for a location to have the same name in two different markets.

Reply:
OK, I had avoided this in the past, but I agree this could work nicely and give us a lot of flexibility with Pages.